cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/DatabaseImpl.java

```diff
@@ -1191,4 +1186,43 @@ public DocumentRevision delete(final DocumentRevision rev) throws
         }
     }
 
+    @Override
+    public void createWithHistory(DocumentRevision revision, int revisionsStart, List<String>
+            revisionsIDs) throws DocumentException {
+
+        Map<String, Attachment> attachments = revision.getAttachments();
+        Collections.list(Collections.enumeration(revision.getAttachments().values()));
+
+        InternalDocumentRevision internalRev = new DocumentRevisionBuilder()
+                .setDocId(revision.getId())
+                .setRevId(revision.getRevision())
+                .setBody(revision.getBody())
+                .setAttachments(attachments)
+                .setDeleted(revision.isDeleted())
+                .build();
+
+        Map<String[], Map<String, PreparedAttachment>> preparedAttachments = Collections
+                .singletonMap
+                        (new String[]{revision.getId(), revision.getRevision()}, AttachmentManager
+                                .prepareAttachments(attachmentsDir, attachmentStreamFactory,
+                                        attachments));
+
+        Map<String, Object> attachmentsMetadata = new HashMap<String, Object>();
```

Comment:
Updated the documentation with more explicit instructions for attachments in 5b6f87db490219631383fbbbc49987d92bb7e297.
Added new attachments tests in 66092c6d007431627cb36cf77e0f2a6030d32355 and while I was in the test class added some additional test doc and improved some assertions in b92d86bb86be54857fb34b1b6347b4a0236355ca.